### PR TITLE
Add support for warningsAsErrors config-option.

### PR DESCRIPTION
**Pull request by josteink**
This partially resolves https://github.com/mozilla/addons-linter/issues/1014.

### CODE REVIEW

File: src/linter.js [hunk 1]
```diff
@@ -426,7 +426,11 @@ export default class Linter {
         // test runs against un-instrumented code.
         /* istanbul ignore if  */
         if (this.config.runAsBinary === true) {
-          process.exit(this.output.errors.length > 0 ? 1 : 0);
+          var exitCode = this.output.errors.length > 0 ? 1 : 0;
```
Comment (tofumatt) [inline src/linter.js]: A nitpick, but we should use `let` here 😄

File: src/linter.js [hunk 2]
```diff
@@ -426,7 +426,11 @@ export default class Linter {
...
+          if (exitCode === 0 && this.config.warningsAsErrors === true) {
+            exitCode = this.output.warnings.length > 0 ? 1 : 0;
```
Comment (tofumatt) [inline src/linter.js]: I think this code might not be checked by our code coverage tool, but could you add a CLI test (I think there are existing ones for guidance) to make sure this works as it should? (Though the code looks fine.)

File: src/linter.js (hunk 1 above)
Comment (josteink) [inline src/linter.js]: Agreed. My bad.

Comment (josteink) [inline src/linter.js]: Agreed. My bad.

File: src/linter.js (hunk 2 above)
Comment (josteink) [inline src/linter.js]: Sounds reasonable. I'll see what I can get done.

**Comment by josteink** [comment]
And there I think I should have addressed all concerns raised so far. Anything else you think needs fixing?

**Comment by josteink** [comment]
@tofumatt Any news on this one?

**Comment by tofumatt** [comment]
It's on my radar but I'm travelling/attending a conference and doing some AMO work right now, so I'll need to get to it after I clear some of that stuff up. Might not be able to review until Tuesday, but I'll try to find an extra moment! 😄

**Comment by josteink** [comment]
Gotcha. And understood.

That said, I have a PR for `web-ext` which depends on this getting merged (which I need for a AMO-related production pipeline I'm building at work), so any time you can find is greatly appreciated :)

**Comment by tofumatt** [comment]
Thanks for your patience, all merged; I'll do a release now as well so you can pull the newest version from npm.

**Comment by josteink** [comment]
Awesome. Thanks for taking in this patch :)

**Comment by atsay** [comment]
Thanks for the patch, Jostein! We've added this contribution to our [recognition page](https://wiki.mozilla.org/Add-ons/Contribute/Recognition#October_2016).

**Pull request merged**